### DISCUSSION TITLE
Testing Discussion

### DISCUSSION THREAD

**Discussion by AdamMcAdamson** [UTDNebula/nebula-api#101] [Q&A]
I want to open the floor for discussion on testing for API. It is a bit of a grey area for me, so I would love to have your input on what kinds of testing we ought to perform and how.

**Comment by ahshabbir** [comment]
After thinking about this more, I think that testing is actually rather complicated since it requires the manufacture of fake data. Usually a QA team would do this in real life. For now, it might be best that we just create a mock database with small amounts of data and test off of that. Then we could test each of the gin handler functions off of those.

**Reply by ahshabbir** [reply]
See this stackoverflow: https://stackoverflow.com/questions/57967604/mocking-official-mongodb-driver

**Reply by ahshabbir** [reply]
If we're ambitious we can use fuzz tests as well, I don't have experience with this but it seems like it would be applicable here:
https://go.dev/doc/fuzz/
